### PR TITLE
Add test for DatePeriod::__set_state

### PR DESCRIPTION
**Pull request by omerida**

### CODE REVIEW

File: ext/date/tests/date_period_set_state_error.phpt [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+--CREDITS--
+Oscar Merida <oscar@musketeers.me>
+#phptestfest2017
+--FILE--
+<?php
+$dp = new \DatePeriod('R4/2012-07-01T00:00:00Z/P7D');
+$dp->__set_state(1);
+?>
+--EXPECTF--
+Warning: DatePeriod::__set_state() expects parameter 1 to be array, %s given in %s on line %d
```
Comment (carusogabriel) [inline ext/date/tests/date_period_set_state_error.phpt:12]: We won't accept ZPP Failures tests :confused:

**Review by carusogabriel — CHANGES_REQUESTED** [review]

**Comment by omerida** [comment]
Is this documented somewhere? There should be a way to exclude these lines from coverage tests if you don’t want tests for them.

**Comment by carusogabriel** [comment]
@omerida Please take a look at https://externals.io/message/103886.

**Comment by petk** [comment]
Every time the pull request here can't get merged a kitten dies and breaks mergers heart, but I also am not sure what other way to do in the time of PHP 7.4... Let not this stop you from adding a new test for something else or with different approach. I'll check what we can do with these if anything...

**Comment by petk** [comment]
Hello, thank you for the pull request. As mentioned above, the ZPP tests aren't being merge to the upstream anymore because they test redundant things. See this discussion for more info at https://externals.io/message/103886

Closing this one as resolved. 👍

I'll also add instructions to qa.php.net site which is basically the main source of information about PHP tests...

**Pull request closed without merging**